dll/win32/shell32/CDefView.cpp

```diff
@@ -1475,6 +1756,105 @@ HRESULT CDefView::CheckViewMode(HMENU hmenuView)
     return S_OK;
 }
 
+LRESULT CDefView::DoColumnContextMenu(LPARAM lParam)
+{
+    const UINT maxItems = 15; // Feels about right
+    const UINT idMore = 0x1337;
+    UINT idFirst = idMore, idLast = idFirst;
+    UINT lastValidListCol = 0; // Keep track of where the new column should be inserted
+    UINT showMore = GetKeyState(VK_SHIFT) < 0;
+    POINT pt = { GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam) };
+    HWND hWndHdr = ListView_GetHeader(m_ListView.m_hWnd);
+
+    if (lParam == ~0)
+    {
+        RECT r;
+        ::GetWindowRect(hWndHdr, &r);
+        pt.x = r.left + ((r.right - r.left) / 2);
+        pt.y = r.top + ((r.bottom - r.top) / 2);
+    }
+
+    HMENU hMenu = CreatePopupMenu();
+    if (!hMenu)
+        return 0;
+
+    for (UINT foldCol = 0;; ++foldCol)
```

Comment:
Place the declaration at the beginning of the function, not within the for-loop.
Here, and in the other locations where you did it like that.

Reply:
Nvm, I overlooked that we are cpp code here. It is ok how you did that.